services/app-api/src/domain-models/contractAndRates/rateTypes.ts

```diff
@@ -1,34 +1,31 @@
 import { z } from 'zod'
+import {
+    contractWithoutDraftRatesSchema,
+    rateWithoutDraftContractsSchema,
+} from './baseContractRateTypes'
 import {
     rateRevisionWithContractsSchema,
     rateRevisionSchema,
 } from './revisionTypes'
-import { statusSchema } from './statusType'
 
-const rateSchema = z.object({
-    id: z.string().uuid(),
-    createdAt: z.date(),
-    updatedAt: z.date(),
-    status: statusSchema,
-    stateCode: z.string(),
-    parentContractID: z.string().uuid(),
-    stateNumber: z.number().min(1),
-    // If this rate is in a DRAFT or UNLOCKED status, there will be a draftRevision
-    draftRevision: rateRevisionWithContractsSchema.optional(),
-    // All revisions are submitted and in reverse chronological order
-    revisions: z.array(rateRevisionWithContractsSchema),
+const baseCategorySchema = z.object({
```

Comment:
what is this stuff?

Reply:
oh whoops that was something from the internet I was using to understand the cycle problem